waiter/src/waiter/core.clj

```diff
@@ -946,17 +947,21 @@
                                                      instance short-circuit? router-ids "blacklist"
                                                      (partial make-blacklist-request make-inter-router-requests-sync-fn blacklist-period-ms)
                                                      reason))))
+   :populate-maintainer-chan-routine! (pc/fnk [[:state instance-rpc-chan]]
+                                        ;; avoid cycle in dependency resolution for functions defined in daemons
```

Comment:
Can you explain this comment a bit more?

Reply:
Why shouldn't everything you've changed to use `:populate-maintainer-chan-routine!` use `:populate-maintainer-chan!` instead?